compiler/rustc_mir_build/src/builder/matches/mod.rs

```diff
@@ -1271,6 +1270,8 @@ pub(crate) struct MatchPairTree<'tcx> {
     /// parent has succeeded. For example, the pattern `Some(3)` might have an
     /// outer match pair that tests for the variant `Some`, and then a subpair
     /// that tests its field for the value `3`.
+    ///
+    /// Invariant: Or-patterns must be sorted to the end.
     subpairs: Vec<Self>,
```

Comment:
Why the invariant on subpairs? I don't think we ever rely on it since we need to re-sort the whole candidate whenever we add new match paris to it.

Reply:
I would prefer to remove this: the invariant is relevant for `Candidate`s only; adding it here is potentially confusing. We could even remove it from `FlatPat` if we make sure to sort in the `FlatPat` -> `Candidate` conversion step.